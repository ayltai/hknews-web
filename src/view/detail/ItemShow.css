.Title {
    font-size: 120% !important;
    font-weight: bold !important;
    padding-bottom: 0.75em;
}
