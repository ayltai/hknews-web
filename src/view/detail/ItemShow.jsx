import React from 'react';
import { ArrayField, DateField, ImageField, RichTextField, SimpleShowLayout, ShowController, ShowView, TextField } from 'react-admin';

import { Gallery } from '../list/Gallery';

const Title = record => (
    <span>{record ? `${record.title}` : ''}</span>
);

export class ItemShow extends React.PureComponent {
    componentDidUpdate = () => {
        window.scrollTo(0, 0);
    }

    render() {
        return (
            <ShowController {...this.props}>
                {controllerProps => (
                    <ShowView
                        title={<Title record={this.props.record} />}
                        {...controllerProps}
                        {...this.props}>
                        <SimpleShowLayout>
                            <RichTextField
                                label=''
                                source='description' />
                            {controllerProps && controllerProps.record && controllerProps.record.images && controllerProps.record.images.length > 0 && (
                                <ArrayField
                                    label=''
                                    source='images'>
                                    <Gallery
                                        source='imageUrl'
                                        caption='description'>
                                        <ImageField
                                            label=''
                                            source='imageUrl'
                                            title='description' />
                                    </Gallery>
                                </ArrayField>
                            )}
                            <TextField
                                label=''
                                source='source.name' />
                            <DateField
                                label=''
                                source='publishDate'
                                showTime />
                        </SimpleShowLayout>
                    </ShowView>
                )}
            </ShowController>
        );
    }
}
