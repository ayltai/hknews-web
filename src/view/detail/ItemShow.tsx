import React from 'react';

import { Gallery } from '../list/Gallery';

const { ArrayField, DateField, ImageField, RichTextField, SimpleShowLayout, ShowController, ShowView, TextField } = require('react-admin');

const Title : ({ record } : any) => React.ReactElement = ({ record } : any) : React.ReactElement => (
    <span>{record ? `${record.title}` : ''}</span>
);

interface IProps {
    record : any;
}

export class ItemShow extends React.PureComponent<IProps, {}> {
    public constructor(props : any) {
        super(props);
    }

    public componentDidUpdate : () => void = () : void => {
        window.scrollTo(0, 0);
    }

    public render() : React.ReactElement {
        return (
            <ShowController {...this.props}>
                {(controllerProps : any) : React.ReactElement => (
                    <ShowView
                        title={<Title record={this.props.record} />}
                        {...controllerProps}
                        {...this.props}>
                        <SimpleShowLayout>
                            <RichTextField
                                label=''
                                source='description' />
                            {controllerProps && controllerProps.record && controllerProps.record.images && controllerProps.record.images.length > 0 && (
                                <ArrayField
                                    label=''
                                    source='images'>
                                    <Gallery
                                        source='imageUrl'
                                        caption='description'>
                                        <ImageField
                                            label=''
                                            source='imageUrl'
                                            title='description' />
                                    </Gallery>
                                </ArrayField>
                            )}
                            <TextField
                                label=''
                                source='source.name' />
                            <DateField
                                label=''
                                source='publishDate'
                                showTime />
                        </SimpleShowLayout>
                    </ShowView>
                )}
            </ShowController>
        );
    }
}
