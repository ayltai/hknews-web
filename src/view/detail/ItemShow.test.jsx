import { mount, } from 'enzyme';
import React from 'react';
import { TestContext, } from 'react-admin';
import { act, } from 'react-dom/test-utils';
import renderer from 'react-test-renderer';

import { ItemShow, } from './ItemShow';

describe('', () => {
    it('', () => {
        expect(renderer.create(
            <TestContext>
                <ItemShow
                    basePath='/'
                    resource='foo' />
            </TestContext>
        ).toJSON()).toMatchSnapshot();
    });

    it('mount without errors', async () => {
        await act(async () => {
            mount(
                <TestContext>
                    <ItemShow
                        basePath='/'
                        resource='foo' />
                </TestContext>
            );
        });
    });
});
