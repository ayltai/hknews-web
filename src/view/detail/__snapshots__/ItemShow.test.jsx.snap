// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="show-page RaShow-root-1"
>
  
  <div
    className="RaShow-main-2 RaShow-noActions-3"
  >
    <div
      className="MuiPaper-root MuiCard-root RaShow-card-4 MuiPaper-elevation1 MuiPaper-rounded"
    />
  </div>
</div>
`;
